code/obj/rocko.dm

```diff
@@ -0,0 +1,184 @@
+			view_chance += 20
+		if(M.hasStatus("drunk"))
+			view_chance += 5
+
+		return prob(view_chance)
+
+	process()
+		if(prob(95))
+			return
+
+		switch(pick( 200;1, 200;2, 50;3, 10;4, 100;5))
+			if(1)
+				emote("<B>[src]</B> winks.", "<I>winks</I>")
+			if(2)
+				if(holder) boutput(holder,"<B>[src]</B> feels warm.")
+			if(3)
+				emote("<B>[src]</B> whispers something about a hellburn.", "<I>whispers something about a hellburn</I>")
+			if(4)
+				emote("<B>[src]</B> rants about job site safety.", "<I>Goes on about job safety</I>")
+			if(5)
+				speak("We really need to do something about the [pick("captain", "head of personnel", "clown", "research director", "head of security", "medical director", "AI")].")
+
+	proc/speak(message)
+		var/list/targets
+		var/image/chat_maptext/chat_text = null
+
+		if(!holder)
+			targets = hearers(src, null)
+			chat_text = make_chat_maptext(src, message, "color: ["#bfd6d8"];", alpha = 200)
+		else
+			targets = list(holder)
+
+		for(var/mob/O in targets)
+			if(src.can_mob_observe(O))
+				O.show_message("<span class='game say bold'><span class='name'>[src.name]</span> says, <span class='message'>\"[message]\"</span></span>", 2, assoc_maptext = chat_text)
+
+	proc/emote(message, maptext_out)
+		var/list/targets
+		var/image/chat_maptext/chat_text = null
+
+		if(!holder)
+			targets = viewers(src, null)
+			chat_text = make_chat_maptext(src, maptext_out, "color: #C2BEBE;", alpha = 120)
+		else
+			targets = list(holder)
+
+		for (var/mob/O in targets)
+			if(src.can_mob_observe(O))
+				O.show_message("<span class='emote'>[message]</span>", assoc_maptext = chat_text)
+
+	proc/update_icon()
+		var/image/smiley = image('icons/misc/rocko.dmi', src.smile ? "smile" : "frown")
+		if(bright)
+			painted = pick(list("#EE2","#2EE", "#E2E","#EEE"))
+		else
+			painted = pick(list("#000","#151","#514","#511","#218"))
+
+		smiley.color = painted
+		smiley.appearance_flags = RESET_COLOR | RESET_ALPHA | RESET_TRANSFORM
+
+		src.UpdateOverlays(smiley, "face")
+		update_hat()
+
+	proc/update_hat()
+		if(istype(src.hat))
+			var/icon/working_icon = icon(src.hat.wear_image_icon, src.hat.icon_state, SOUTH )
+			working_icon.Shift(SOUTH, 10)
+			var/image/working_hat = image(working_icon)
+			working_hat.appearance_flags = RESET_COLOR | RESET_ALPHA | RESET_TRANSFORM
+			src.UpdateOverlays(working_hat, "hat")
+		else
+			src.UpdateOverlays(null, "hat")
+
+	get_desc(dist, mob/user)
+		if(ismob(user) &&	user.job == "Chief Engineer")
+			. = "A rock but also [pick(rocko_is)]."
+		else if(ismob(user) && (user.job in list("Engineer", "Mechanic", "Quartermaster", "Captain")))
+			. = "The Chief Engineer loves this rock.  Maybe it's to make up for their lack of a pet."
+		else
+			. = "A rock with a [src.smile ? "smiley" : "frowny"] face painted on it."
+
+	attack()
```

Comment:
Empty proc?

Reply:
Could use a return, to clarify it's an override

Reply:
Uh, that shouldn't be there.